src/builtin-addons/core/script-definition-provider.ts

```diff
@@ -185,12 +194,23 @@ export default class CoreScriptDefinitionProvider {
     } else if (isImportSpecifier(astPath)) {
       logInfo(`Handle script import for Project "${project.name}"`);
       const pathName: string = ((astPath.parentFromLevel(2) as unknown) as t.ImportDeclaration).source.value;
+      const pathParts = pathName.split('/');
+      let maybeAppName = pathParts.shift();
 
-      project.roots.forEach((projectRoot) => {
-        const potentialPaths = this.guessPathForImport(projectRoot, uri, pathName) || [];
+      if (maybeAppName && maybeAppName.startsWith('@')) {
+        maybeAppName = maybeAppName + path.sep + pathParts.shift();
+      }
 
-        definitions = definitions.concat(potentialPaths);
-      });
+      let potentialPaths: Location[];
+
+      // If the start of the pathname is same as the project name, then use that as the root.
+      if (project.name === maybeAppName) {
+        potentialPaths = this.guessPathForImport(project.root, uri, pathParts.join('/')) || [];
+      } else {
+        potentialPaths = this.guessPathForImport(project.root, uri, pathName) || [];
```

Comment:
should we search here for project.addonsMeta.name?

Reply:
i dont think that would be needed since this is mainly to get the root of your project and not anything for addons